.github/workflows/ci.yml

```diff
@@ -31,11 +36,12 @@ jobs:
   upload:
     name: Upload to GCS
     needs: [build]
+    if: github.event_name == 'push'
     uses: ./.github/workflows/upload.yml
     secrets: inherit
     with:
       version: ${{ github.event.pull_request.head.sha || github.sha }}
-      github_environment: ${{ github.event_name != 'push' && 'gcs' || 'gcs-no-approval' }}
+      github_environment: gcs-no-approval
```

Comment:
cc @joey-grafana

> Can you please elaborate on

> [!IMPORTANT]
> Before merging to main update restriction rules for gcs-no-approval 

Before this change we used:
- "gcs" GitHub environment for pushes to branches (configured only for PRs)
- "gcs-no-approval" for main and tags (so it runs automatically)

With this change we will use "gcs-no-approval" for all, including PRs

However "gcs-no-approval" has a restriction rule that it can be applied only on "main" and "v*" branches. When we merge this PR we can remove the restriction.

Anyone who can add a label will be able to upload the artifact = anyone who can create a PR in the main repo (not forks)

Reply:
For context: spoke with Piotr who mentioned only Grafana users can trigger an upload from the label 👍